tools/autograd/derivatives.yaml

```diff
@@ -2020,8 +2020,8 @@
   result: auto_element_wise
 
 - name: threshold_(Tensor(a!) self, Scalar threshold, Scalar value) -> Tensor(a!)
-  self: threshold_backward(grad, result, threshold)
-  result: auto_element_wise
+  self: threshold_backward(grad, self, threshold)
```

Comment:
That's not ok I think.
If you do `threshold(self, a, b)` with `b > a`, this will be wrong!

Reply:
Oh I think that was happening before this change, but not anymore? 
We want to do `grad_input = self > threshold ? grad_output : zero` right?

Reply:
1. We should have such a case in the OpInfo samples
2. We want to check self yes IIRC (use OpInfo testing to see which one is correct to be sure)
3. Changing this can have significant impact on memory usage patterns. This is the inplace relu op that is used everywhere. Using `self` here means that it will be cloned: extra buffer!

Reply:
> 1. We should have such a case in the OpInfo samples
> 2. We want to check self yes IIRC (use OpInfo testing to see which one is correct to be sure)

We don't have an explicit check because all values are chosen at random, but that means it will be checked with high probability.

> 3. Changing this can have significant impact on memory usage patterns. This is the inplace relu op that is used everywhere. Using self here means that it will be cloned: extra buffer!

Okay I checked and relu doesn't actually use threshold (and since a == b for relu, so we can still be better off saving result instead of self). For threshold though I guess that now it makes less sense to use in-place (at least when gradients are required).